src/gui/TextEditor.cpp

```diff
@@ -904,8 +905,66 @@ auto TextEditor::blinkCallback(TextEditor* te) -> gint {
     return false;
 }
 
+void TextEditor::setTextToPangoLayout(PangoLayout* pl) const {
+    if (!this->preeditString.empty()) {
+        GtkTextIter cursorIter = {nullptr};
+        GtkTextMark* cursor = gtk_text_buffer_get_insert(this->buffer);
+        gtk_text_buffer_get_iter_at_mark(this->buffer, &cursorIter, cursor);
+
+        string text = this->text->getText();
+        int offset = gtk_text_iter_get_offset(&cursorIter);
+        int pos = gtk_text_iter_get_line_index(&cursorIter);
+
+        for (gtk_text_iter_set_line_index(&cursorIter, 0); gtk_text_iter_backward_line(&cursorIter);) {
```

Comment:
```suggestion
        // Calculate the byte offset into the text contents. Since gtk doesn't provide a method to do this, we need to manually count bytes per line.
        for (gtk_text_iter_set_line_index(&cursorIter, 0); gtk_text_iter_backward_line(&cursorIter);) {
```